internal/printer/print-to-js.go

```diff
@@ -303,12 +303,11 @@ func render1(p *printer, n *Node, opts RenderOptions) {
 		}
 
 		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			p.addSourceMapping(c.Loc[0])
 			if c.Type == TextNode {
-				p.addSourceMapping(c.Loc[0])
 				p.print(c.Data)
 			} else {
-				p.addSourceMapping(c.Loc[0])
-				if (c.PrevSibling == nil || (c.PrevSibling != nil && c.PrevSibling.Type == TextNode)) && c.NextSibling != nil {
+				if c.PrevSibling == nil || (c.PrevSibling != nil && c.PrevSibling.Type == TextNode) {
```

Comment:
This was the expression I removed: `c.NextSibling != nil`. Not sure if that was a logical bug, or if I just introduced a regression. But the test added seemed to pass now.

Reply:
I feel like I ran into a condition where that was needed, but I can't remember what it was.